### ISSUE TITLE
Potential regression with makeBindingTarget

### ISSUE THREAD

**Issue by KingOfTheHuns** [ReactiveCocoa/ReactiveCocoa#3350]
Hi,

It seems that there is a regression in the latest version of ReactiveCocoa.
I'm trying to use the <~ operator between a UILabel object and a Property<String> and it seems that it doesn't work anymore as previously.

```
label.reactive.text <~ viewModel.statusText // label is retained UILabel object statusText is a Property<String> object
```

If you take a look at to the code of ```makeBindingTarget```

```
public func makeBindingTarget<U>(on scheduler: SchedulerProtocol = UIScheduler(), _ action: @escaping (Base, U) -> Void) -> BindingTarget<U> {
		return BindingTarget(on: scheduler, lifetime: lifetime) { [weak base = self.base] value in
			if let base = base {
				action(base, value)
			}
		}
	}
```

It seems that the returned BindingTarget object from label.reactive.text is deallocated immediately.

If I do this:

```
textBindingTarget = label.reactive.text // textBindingTarget is retained member
textBindingTarget <~ viewModel.statusText // label is retained UILabel object statusText is a Property<String> object
```

then it works. This behaviour is new in the earlier version of ReactiveCocoa there was no need to retain the BindingTarget object returned from ```<UIKitClass>.reactive.<property>```.

Is this a new feature or a bug?

Any help is appreciated

Br,
Attila

**Comment by andersio** [comment]
This shouldn't happen.

/cc @mdiep

**Comment by andersio** [comment]
The binding operator is behaving as expected though.

May I ask if `Property<String>` is an existential? You might need to be aware of the fact that `Property.init(_:)` no longer retains the property it wraps. [`Property(capturing:)` is introduced for this](https://github.com/ReactiveCocoa/ReactiveSwift/releases/tag/1.0.0-rc.1):

> `Property.init(_:)` now creates a property which observes the given property without retaining it. The existential property initializer is now renamed to `Property.init(capturing:)`, and is subject to deprecation when generalised existentials land.

**Comment by andersio** [comment]
Oh, it is a bug in ReactiveSwift that happens only when the parameter of `BindingTarget` is an optional. Will push a fix ASAP.

**Comment by KingOfTheHuns** [comment]
@andersio Thanks a bunch for fixing the bug. I haven't tested it yet but I will once the Cartfile of ReactiveCocoa is updated to use the latest ReactiveSwift. :)

**Comment by andersio** [comment]
Fixed in 5.0.0-alpha.5.
